src/lib/rest/curl_context.cpp

```diff
@@ -0,0 +1,94 @@
+#include "curl_context.h"
+
+#include <map>
+
+#include <assert.h>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+
+static pthread_mutex_t contexts_mutex = PTHREAD_MUTEX_INITIALIZER;
+static std::map<std::string, struct curl_context> contexts;
+
+struct curl_context
+get_curl_context(const std::string& key)
+{
+  struct curl_context cc = {NULL, NULL};
+  int s = pthread_mutex_lock(&contexts_mutex);
+  if(s!=0)
+  {
+       LM_X(1,("pthread_mutex_lock"));
```

Comment:
I think many libraries have assert for "Impossible" conditions and many errors are so serious that there is no other reasonable behavior other than  exiting immediately.  I can't think on a sane case where a mutex can't  be initialized or be locked and we should continue happily.